### DISCUSSION TITLE
POWER Fail and EEPROM on tiny806

### DISCUSSION THREAD

**Discussion by coderatchance** [SpenceKonde/megaTinyCore#1091] [Q&A]
Hello all,

I am wondering how to tackle the power fail with regard to data loos prevention.
For example, how to update the value in the eeprom when the tiny looses VDD.
Was thinking to go via the BOD, for example, when it fires it updates the eeprom, 
however, never done that before, do you have ideas how that is usually done? 

Any ideas are very appreciated !

**Comment by mechatroniks-git** [comment]
What voltage are you sending to the ATtiny? Where is that voltage rail being made? 

If for example you are regulating 5V from 12V, then sense the 12V through a resistor divider and put a properly sized bulk cap on the 5V, to buy you enough time to complete the write. 

Be sure to use the AC/comparator to detect a downward trending voltage, trigger interrupt and write flash.

**Comment by coderatchance** [comment]
Thanks @mechatroniks-git , I thought about that as well, but it is not my design, so I cannot really delve into HW, anyway, I am on 3.3V.
Indeed, writing to eeprom takes some time, so, the solution so far is update the eeprom, and it should be ok for another 25 years or so, I am not even bothering thinking about it :)
